packages/@lwc/engine-core/src/framework/modules/attrs.ts

```diff
@@ -5,26 +5,28 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
 import { assert, isNull, isUndefined, keys, StringCharCodeAt } from '@lwc/shared';
+
 import { setAttribute, removeAttribute } from '../../renderer';
+import { VElement } from '../../3rdparty/snabbdom/types';
+
 import { unlockAttribute, lockAttribute } from '../attributes';
 import { EmptyObject } from '../utils';
-import { VElement } from '../../3rdparty/snabbdom/types';
 
 const xlinkNS = 'http://www.w3.org/1999/xlink';
 const xmlNS = 'http://www.w3.org/XML/1998/namespace';
 const ColonCharCode = 58;
 
-function updateAttrs(oldVnode: VElement, vnode: VElement) {
+export function patchAttributes(oldVnode: VElement | null, vnode: VElement) {
     const {
         data: { attrs },
     } = vnode;
 
     if (isUndefined(attrs)) {
         return;
     }
-    let {
-        data: { attrs: oldAttrs },
-    } = oldVnode;
+
+    let oldAttrs = isNull(oldVnode) ? undefined : oldVnode.data.attrs;
```

Comment:
This is a bit confusing. If `attrs` is undefined, then we've already returned on line 25, so there's no need to set `oldAttrs` to undefined and compare it to `attrs`.

I know this is equivalent to the old code; I just wonder if maybe we need a code comment or something here to explain it, or to refactor the if/else logic. E.g.:

```js
if (!isNull(oldVnode) && oldVnode.data.attrs === attrs) {
  return
}
```

Reply:
I refactored the code, by setting `oldAttrs` to an `EmptyObject` instead of `undefined`. Let me know if it makes it more readable.